### ISSUE TITLE
`prefer-spread` to allow `Array.from()` with `Map.prototype.entries()`

### ISSUE THREAD

**Issue by yuan116** [sindresorhus/eslint-plugin-unicorn#2445]
### Description

Hi, i just started to use this plugin, and found out `prefer-spread` showing error to me on this below sample code

```js
const map = new Map();
// some data add into map
const entries = Array.from(map.entries());
// add data into specific index
entries.splice(
    entries.findLastIndex(([_id, _value]) => {
        return _id === value.parentId;
    }) + 1,
    0,
    [id, value]
)
const map2 = new Map(entries);
```

I have more than 100 data in a Map object, `[...map.entries()]` seem to be slower than `Array.from(map.entries())` when it has 50 data in it
below are the online js benchmark platform to test
1. [jsben](https://jsben.ch/XBkwq)
2. [jsbenchmark](https://jsbm.dev/Zd7Ky1qAtbKNA)

My suggestion is to allow the use of `Array.from()` with `Map.prototype.entries()` when `prefer-spread` is on `(error/warn)`

### Fail

```js
const map = new Map();
const entries = Array.from(map.entries());
```

### Pass

```js
const map = new Map();
const entries = Array.from(map.entries());
```

**Comment by sindresorhus** [comment]
https://sindresorhus.com/blog/micro-benchmark-fallacy

**Comment by yuan116** [comment]
hmm, ok

i understand what you trying to say

but personally still hope it allow `Array.from()` with `map` (Map object) or `Map.prototype.entries()`

you may close this issue, if you think this issue is not reasonable
